File: railway.judge.py

from Judger import Judger
import sys
import numbers


class Tester(Judger): 
    def __init__(self, debug=True):
        if debug:
            super().__init__("railway.json")
        else:
            super().__init__(sys.argv[1], debug=False, save=True, clean_after_read=True)

    def run(self, sample):
        import imp
        Railway = imp.load_source("Railway", 'railway.sol.py').Railway
        out = Railway().railway(sample['landmarks'], sample['distance'])
        return out

    def compare(self, out, sample):
        if type(out) is not int:
            self.debugPrint('>', sample)
            self.debugPrint('<', out)
            return False
        answer = sample['answer']
        # print(answer)
        if answer != out:
            self.debugPrint('>', sample)
            self.debugPrint('<', out)
        return answer == out


Tester(debug=True).judge()
# Tester(debug=True).judge()

File: railway.judge.java
import com.google.gson.JsonElement;
import com.google.gson.JsonArray;
import com.google.gson.reflect.TypeToken;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Collections;
import java.lang.Math;
/*
dk openjdk:14-slim javac -cp gson.jar:algs4.jar:. railway.sol.java railway.judge.java Judger.java
dk openjdk:14-slim java -cp gson.jar:algs4.jar:. Judge
*/

class Judge extends Judger<Integer> {

    public Judge(boolean debug, boolean clean_after_read) {
        super(debug);
        this.file_path = "railway.json";
        this.clean_after_read = clean_after_read;
    }

    static class Data {
        List<int[]> distance;
        int landmarks;
        Integer answer;
        public Data() {};
    }

    @Override protected boolean compare(Integer out, JsonElement ele) {
        Data d = this.gson.fromJson(ele, Data.class);
        return d.answer.equals(out);
    }

    @Override protected Integer run(JsonElement ele) {
        Data d = this.gson.fromJson(ele, Data.class);
        this.resetTime();
        return new Railway().railway(d.landmarks, d.distance);
    }

    @Override protected void debugPrint(Integer out, JsonElement ele) {
        System.out.print(">");
        System.out.println(this.gson.fromJson(ele, Data.class).answer);
        System.out.print("<");
        System.out.println(out);
    };

    public static void main(String []args) {
        // Judge j = new Judge(false, true);
        Judge j = new Judge(true, false);
        j.judge(args);
    }
}
